### DISCUSSION TITLE
How to change whole URL in cy.request

### DISCUSSION THREAD

**Discussion by Lumca** [cypress-io/cypress#17453] [Questions and Help]
Hi,
I need to send a request on specified URL, but it always adds previous visited URL.
Lets say I need to send request to ws://localhost:8580/

The way the test works in this order:
1. Logins to app
2. Visits homepage for example google.com
3. Cypress sends POST request to ws://localhost:8580/

What happens is the request is not send to localhost:8580 but it is send to google.com/ws://localhost:8580/.
How do I prevent cypress from doing that?

**Comment by vaclavGabriel** [comment]
Hello Lumca. 
Have you tried to use an absolute URL in the `cy.request()` command? 

`cy.request({
method: "POST", 
url: "ws://localhost:8580/"
})`

**Reply by Lumca** [reply]
Hello @vaclavGabriel,
thanks for response and yes I have try that.

**Reply by sandra-ouadghiri** [reply]
Maybe ws is not seen as a valid protocol (and thus your arg is seen as part of a relative url, not an absolute). Try a different protocol (ex http) just for the test. If it works, then maybe raise a ticket to ask for the ws protocol support ? Also, for url question it's best to precise what's your baseUrl for Cypress (you can log Cypress.config('baseUrl') or look in the UI Cypress config tab).
